packages/engine/src/avatar/functions/autopilotFunctions.ts

```diff
@@ -0,0 +1,28 @@
+import { Vector3 } from 'three'
+
+import { Engine } from '../../ecs/classes/Engine'
+import { Entity } from '../../ecs/classes/Entity'
+import { Physics, RaycastArgs } from '../../physics/classes/Physics'
+import { CollisionGroups } from '../../physics/enums/CollisionGroups'
+import { getInteractionGroups } from '../../physics/functions/getInteractionGroups'
+import { SceneQueryType } from '../../physics/types/PhysicsTypes'
+
+export const autopilotGetPosition = (entity: Entity): Vector3 | undefined => {
+  const physicsWorld = Engine.instance.currentWorld.physicsWorld
+  const world = Engine.instance.currentWorld
+  const interactionGroups = getInteractionGroups(CollisionGroups.Default, CollisionGroups.Ground)
+  const raycastArgs = {
```

Comment:
this object can be moved out of the function and we update the properties that need to be updated